include/envoy/http/filter.h

```diff
@@ -495,19 +495,24 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    */
   virtual uint32_t decoderBufferLimit() PURE;
 
-  // Takes a stream, and acts as if the headers are newly arrived.
-  // On success, this will result in a creating a new filter chain and likely upstream request
-  // associated with the original downstream stream.
-  // On failure, if the preconditions outlined below are not met, the caller is
-  // responsible for handling or terminating the original stream.
-  //
-  // This is currently limited to
-  //   - streams which are completely read
-  //   - streams which do not have a request body.
-  //
-  // Note that HttpConnectionManager sanitization will *not* be performed on the
-  // recreated stream, as it is assumed that sanitization has already been done.
-  virtual bool recreateStream() PURE;
+  /**
+   * Takes a stream, and acts as if the headers are newly arrived.
+   * On success, this will result in a creating a new filter chain and likely
+   * upstream request associated with the original downstream stream. On
+   * failure, if the preconditions outlined below are not met, the caller is
+   * responsible for handling or terminating the original stream.
+   *
+   * This is currently limited to
+   *   - streams which are completely read
+   *   - streams which do not have a request body.
+   *
+   * Note that HttpConnectionManager sanitization will *not* be performed on the
+   * recreated stream, as it is assumed that sanitization has already been done.
+   *
+   * @param headers Headers used for logging in the access logs and for charging stats. Ignored if
+   * null.
+   */
+  virtual bool recreateStream(const ResponseHeaderMap* headers) PURE;
```

Comment:
I would call this `original_response_headers` or something, at a glance I thought this was the request headers to recreate the stream with

Reply:
Good point - fixed now!